File: app/src/main/java/com/ihadzhi/eatlimination/data/Converters.java
package com.ihadzhi.eatlimination.data;

import androidx.room.TypeConverter;

import com.ihadzhi.eatlimination.data.SymptomRecord.SymptomCategory;

import java.util.Date;

public class Converters {

    @TypeConverter
    public static Date fromTimestamp(Long value) {
        return value == null ? null : new Date(value);
    }

    @TypeConverter
    public static Long dateToTimestamp(Date date) {
        return date == null ? null : date.getTime();
    }

    @TypeConverter
    public static SymptomCategory fromInt(int index) {
        return index >= 0 && index < SymptomCategory.values().length ? null : SymptomCategory.values()[index];
    }

    @TypeConverter
    public static Integer categoryToInt(SymptomCategory category) {
        return category == null ? null : category.ordinal();
    }

}

File: app/src/main/java/com/ihadzhi/eatlimination/data/Converters.kt
package com.ihadzhi.eatlimination.data

import androidx.room.TypeConverter
import com.ihadzhi.eatlimination.data.SymptomRecord.SymptomCategory
import java.util.*

object Converters {
    @JvmStatic
    @TypeConverter
    fun fromTimestamp(value: Long?): Date? {
        return if (value == null) null else Date(value)
    }

    @JvmStatic
    @TypeConverter
    fun dateToTimestamp(date: Date?): Long? {
        return date?.time
    }

    @JvmStatic
    @TypeConverter
    fun fromInt(index: Int): SymptomCategory? {
        return if (index >= 0 && index < SymptomCategory.values().size) SymptomCategory.values()[index] else null
    }

    @JvmStatic
    @TypeConverter
    fun categoryToInt(category: SymptomCategory?): Int? {
        return category?.ordinal
    }
}